libs/inform/src/lib/services/modal/modal.service.ts

```diff
@@ -0,0 +1,234 @@
+import { Inject, Injectable, Optional, Type } from '@angular/core';
+import { Dialog } from '@angular/cdk/dialog';
+import {
+	combineLatest,
+	filter,
+	map,
+	NEVER,
+	Observable,
+	startWith,
+	Subject,
+	takeUntil,
+	tap,
+} from 'rxjs';
+
+import { NgxModalConfiguration, NgxModalOptions } from '../../types';
+import { NgxModalConfigurationToken } from '../../tokens';
+import { NgxModalAbstractComponent } from '../../abstracts';
+
+/**
+ * A wrapper service to Angular CDK Dialog that provides a WCAG/ARIA compliant implementation of modals
+ *
+ * @export
+ * @class NgxModalService
+ */
+@Injectable({ providedIn: 'root' })
+export class NgxModalService {
+	/**
+	 * A subject to hold the close event
+	 */
+	private modalClosedSubject: Subject<void>;
+
+	constructor(
+		@Optional()
+		@Inject(NgxModalConfigurationToken)
+		private readonly configuration: NgxModalConfiguration,
+		private readonly dialogService: Dialog
+	) {}
+
+	/**
+	 * Opens a modal based on the provided options
+	 *
+	 * @param {NgxModalOptions<ActionType>} options - The modal options
+	 */
+	public open<ActionType extends string = string>(
+		options: NgxModalOptions<ActionType>
+	): Observable<ActionType> {
+		this.modalClosedSubject = new Subject();
+
+		// Iben: Fetch the type of component we wish to show
+		const configuration = this.configuration?.modals?.[options.type];
+		const component: Type<NgxModalAbstractComponent<ActionType>> =
+			options.component ||
+			(this.configuration?.modals?.[options.type].component as Type<
+				NgxModalAbstractComponent<ActionType>
+			>);
+
+		// Iben: Check if all the correct parameters are set and return NEVER when they're not correctly set
+		if (!this.runARIAChecks<ActionType>(options, component)) {
+			return NEVER;
+		}
+
+		// Iben: Render the modal
+		const modal = this.createModalComponent<ActionType>(options, component);
+
+		// Iben: Return the modal action
+		return combineLatest([
+			// Iben: Set the start value to undefined so both actions at least emit once
+			modal.action.pipe(startWith(undefined)),
```

Comment:
I think we've discussed this in the past, but just noting that using `undefined` is only considerd when using ES5+ in strict mode. If not, `undefined` can be overwritten and this can cause unwanted behaviour.

(Also applies to several other places in this PR)

Reply:
I am aware, but given that this is just internal use I don't see an issue here in this specific spot? Unless you can point me to a potential issue here?

For me it's the same as using `first` vs `take(1)` in rjxs. As long as you know the difference and you use the correct one in the correct situation, I'm fine with either one if you use it in a situation where they're interchangeable.